lib/turnsole/heliotrope/service.rb

```diff
@@ -290,6 +306,78 @@ def unsubscribe_product_institution(product_identifier:, institution_identifier:
         response.success?
       end
 
+      #
+      # Access { :full, :read, :none }
+      #
+      #   :full - both read and download access
+      #   :read - read access (no download access)
+      #   :none - no access, but has subscription
+      #
+      #   :undefined - no subscription
+      #
+
+      def product_individual_access?(product_identifier:, individual_identifier:, access_type:)
+        return false unless PRODUCT_ACCESS_TYPES.include?(access_type)
+
+        access_type == product_individual_access(product_identifier: product_identifier, individual_identifier: individual_identifier)
+      end
+
+      def product_individual_access(product_identifier:, individual_identifier:)
+        product_id = find_product(identifier: product_identifier)
+        id = find_individual(identifier: individual_identifier)
+        return :undefined if product_id.nil? || id.nil?
+
+        response = connection.get("products/#{product_id}/individuals/#{id}/access")
+        return :undefined unless response.success?
+
+        response.body["access"].to_sym
+      end
+
+      def access_product_individual(product_identifier:, individual_identifier:, access_type:)
+        return false unless PRODUCT_ACCESS_TYPES.include?(access_type)
+
+        product_id = find_product(identifier: product_identifier)
+        id = find_individual(identifier: individual_identifier)
+        return false if product_id.nil? || id.nil?
+
+        response = connection.post("products/#{product_id}/individuals/#{id}/access", { access: access_type }.to_json)
+        return true if response.success?
+
+        warn "Failed to set access #{access_type} on product #{product_identifier} for individual #{individual_identifier}"
+        false
+      end
+
+      def product_institution_access?(product_identifier:, institution_identifier:, access_type:)
+        return false unless PRODUCT_ACCESS_TYPES.include?(access_type)
+
+        access_type == product_institution_access(product_identifier: product_identifier, institution_identifier: institution_identifier)
+      end
+
+      def product_institution_access(product_identifier:, institution_identifier:)
+        product_id = find_product(identifier: product_identifier)
+        id = find_institution(identifier: institution_identifier)
+        return :undefined if product_id.nil? || id.nil?
+
+        response = connection.get("products/#{product_id}/institutions/#{id}/access")
+        return :undefined unless response.success?
+
+        response.body["access"].to_sym
+      end
+
+      def access_product_institution(product_identifier:, institution_identifier:, access_type:)
```

Comment:
I don't understand the difference between this method and [subscribe_product_institution](https://github.com/mlibrary/turnsole/pull/14/files#diff-b0b53bcfa5d13675f7277c74c07a8fc8b794306004be1c6354ea1d2d6a528cc0R283)

Is the difference that this method is changing an access type on an existing subscription? Whereas the other method is creating a new subscription and setting its access type?

I think their names need to change to something more illustrative, to make the distinction clearer.